tls/s2n_recv.c

```diff
@@ -73,6 +56,25 @@ int s2n_read_full_record(struct s2n_connection *conn, uint8_t * record_type, int
         }
         conn->wire_bytes_in += r;
     }
+
+    return S2N_SUCCESS;
+}
+
+int s2n_read_full_record(struct s2n_connection *conn, uint8_t * record_type, int *isSSLv2)
+{
+    *isSSLv2 = 0;
+
+    /* If the record has already been decrypted, then leave it alone */
+    if (conn->in_status == PLAINTEXT) {
+        /* Only application data packets count as plaintext */
+        *record_type = TLS_APPLICATION_DATA;
+        return 0;
+    }
+    GUARD(s2n_stuffer_resize_if_empty(&conn->in, S2N_LARGE_FRAGMENT_LENGTH));
```

Comment:
Looks like we only previously resized the stuffer for the header, not for the body. What would we gain from always resizing? Is the idea that we allocate the memory all at once instead of potentially on each read attempt?

Reply:
When an HTTP connection becomes idle, we can release all the buffers which have been completely consumed, reducing the memory overhead of maintaining persistent connections. So the resize allows us to then reuse the `s2n_connection` object again at a later time without deleting/creating a new one.

We would need to think more about how this plays in QUIC, so we can leave as is for now